### PR TITLE
Fixes issue #3457 of duplicate column names in passengerPerTripSubway.csv when the data contains only zeroes.

### LINKED ISSUE #3457: passengerPerTripSubway.csv contains duplicate columns
When the data for `0.passengerPerTripSubway.csv` is empty (meaning only zero numbers), then there are duplicate columns (header=1-0) in the file.

image

We should fix this and only print 1 column.

### PR DESCRIPTION
**Pull request by HoneyTauOverTwo**
Assuming that this is the [desired output](https://github.com/LBNL-UCB-STI/beam/issues/3457#issuecomment-1242211276).

IsDesirableOutput

- - -
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3620)

### CODE REVIEW

**Comment by HoneyTauOverTwo** [comment]
As requested by @nikolayilyin, made changes to omit the columns that would be filled with zeroes.

This is what the empty data results looks like:
outputOmmited0

And I also modified the input data [Map<Integer, Map<Integer, Integer>>](https://github.com/LBNL-UCB-STI/beam/blob/develop/src/main/java/beam/analysis/plots/passengerpertrip/GenericPassengerPerTrip.java#L111) on debug to test the intermediate cases (with 1 - 3 non-null columns), sample result for two:

outputOmmited1

This code only affects the output for bus, subway and bike (at least for the beamville data set). The other transport modes (car and ride hail) remain unchanged.

**Comment by HoneyTauOverTwo** [comment]
Removed columns filled with zeroes from passengerPerTripCar as well.

passengerPerTripCar

I figured the way I implemented for buses and subways can possibly have zeroed columns. I am only omitting the last X zeroed columns.

**Comment by nikolayilyin** [comment]
test!

File: src/test/scala/beam/sim/output/IterationsPassengerPerTripTests.scala [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+import scala.collection.mutable.ListBuffer
+import scala.util.control.Breaks.{break, breakable}
+
+class IterationsPassengerPerTripTests
+  extends AnyWordSpecLike
+    with Matchers
+    with BeamHelper
+    with BeforeAndAfterAllConfigMap
+    with Retries {
```
Comment (nikolayilyin) [inline src/test/scala/beam/sim/output/IterationsPassengerPerTripTests.scala]: it seems there is no need in most of these extensions

**Review by nikolayilyin — APPROVED** [review]
looks good

**Comment by nikolayilyin** [comment]
test!

File: src/test/scala/beam/sim/output/IterationsPassengerPerTripTests.scala (hunk 1 above)
Comment (HoneyTauOverTwo) [reply src/test/scala/beam/sim/output/IterationsPassengerPerTripTests.scala]: Oh, certainly... I copied BeamWarmStartRunSpec.scala and missed removing this.

**Comment by nikolayilyin** [comment]
test!

File: src/test/scala/beam/utils/SnapCoordinateSpec.scala [hunk 2]
```diff
@@ -67,7 +69,8 @@ class SnapCoordinateSpec extends AnyWordSpec with Matchers with BeamHelper {
   }
 
   "scenario plan" should {
-    "contain all valid locations" in {
+    // TODO: Investigate, There might be an issue hidden in this flaky test. Retryable tag was added to pass an unrelated PR
```
Comment (HoneyTauOverTwo) [inline src/test/scala/beam/utils/SnapCoordinateSpec.scala:72]: I added this comment to show this was not investigated. Just added the Retryable tag. I found two flaky tests on this file.

**Comment by HoneyTauOverTwo** [comment]
test!

File: src/test/scala/beam/utils/SnapCoordinateSpec.scala (hunk 2 above)
Comment (nikolayilyin) [reply src/test/scala/beam/utils/SnapCoordinateSpec.scala:72]: I think all the tests marked as Retryable deserves some investigation, no need to mark them individually

**Comment by HoneyTauOverTwo** [comment]
test!

**Review by nikolayilyin — APPROVED** [review]
lgtm

**Pull request merged**